src/cloud-sql-instance.ts

```diff
@@ -107,51 +115,118 @@ export class CloudSQLInstance {
   }
 
   async refresh(): Promise<void> {
+    const currentRefreshId = this.scheduledRefreshID;
+
     // Since forceRefresh might be invoked during an ongoing refresh
     // we keep track of the ongoing promise in order to be able to await
     // for it in the forceRefresh method.
     // In case the throttle mechanism is already initialized, we add the
     // extra wait time `limitRateInterval` in order to limit the rate of
     // requests to Cloud SQL Admin APIs.
-    this.ongoingRefreshPromise = this.throttle
-      ? this.throttle(this._refresh).call(this)
-      : this._refresh();
-
-    // awaits for the ongoing promise to resolve, since the refresh is
-    // completed once the promise is resolved, we just free up the reference
-    // to the promise at this point, ensuring any new call to `forceRefresh`
-    // is able to trigger a new refresh
-    await this.ongoingRefreshPromise;
-    this.ongoingRefreshPromise = undefined;
+    this.ongoingRefreshPromise = (
```

Comment:
Instead of ongoingRefreshPromise, could we use `current` and `next` to match all the other connectors?

Reply:
I believe `current` and `next` refer to the refresh result values? 

This property references the ongoing `performRefresh` promise (to be clear, the promise object itself, not its resulting values) - so maybe `performRefreshPromise` is a possible better name?

Reply:
oh I see, in this case it does make sense to update the name to `next` in order to match that!